dotcom-rendering/src/server/render.article.apps.tsx

```diff
@@ -34,8 +38,30 @@ export const renderArticle = (
 		</ConfigProvider>,
 	);
 
-	const clientScripts = [getPathFromManifest('client.apps', 'index.js')];
-	const scriptTags = generateScriptTags([...clientScripts].filter(isString));
+	// We want to only insert script tags for the elements or main media elements on this page view
+	// so we need to check what elements we have and use the mapping to the the chunk name
+	const elements: FEElement[] = article.blocks
+		.map((block) => block.elements)
+		.flat();
+	const pageHasTweetElements = elements.some(
+		(element) =>
+			element._type ===
+			'model.dotcomrendering.pageElements.TweetBlockElement',
+	);
+	const pageHasNonBootInteractiveElements = elements.some(
+		(element) =>
+			element._type ===
+				'model.dotcomrendering.pageElements.InteractiveBlockElement' &&
+			element.scriptUrl !==
+				'https://interactive.guim.co.uk/embed/iframe-wrapper/0.1/boot.js', // We have rewritten this standard behaviour into Dotcom Rendering
+	);
+
+	const clientScripts = [
+		getPathFromManifest('client.apps', 'index.js'),
+		pageHasNonBootInteractiveElements &&
+			`${ASSET_ORIGIN}static/frontend/js/curl-with-js-and-domReady.js`,
```

Comment:
Makes sense! This is what [web](https://github.com/guardian/dotcom-rendering/blob/ee6aa9d4eb5b1027012319d4f466bed9f189a99e/dotcom-rendering/src/server/render.article.web.tsx#L107-L108) does!